### PR TITLE
Fixes #19554 - Provide warning for unsupported PXE loader combinations

### PR DESCRIPTION
**Pull request by tstrachota**

### CODE REVIEW

**Comment by tstrachota** [comment]
@lzap would you mind doing the review?

File: webpack/assets/javascripts/foreman_hosts.js [hunk 1]
```diff
@@ -31,3 +31,97 @@ export function getAttributesToPost(componentType) {
...
+      return null;
+    }
+  )
+);
+
+// RHEL 6.x and Grub1
+// RHEL 7.x and Grub2
+pxeCompatibility.set('rhel',
+  new PXECompatibilityCheck(
+    /(?:red[ ]*hat|rhel|cent[ ]*os)[^\d]*(\d+)(?:[.]\d+)?/,
```
Comment (lzap) [inline webpack/assets/javascripts/foreman_hosts.js]: It is worth including "scientific" and "oracle".

File: webpack/assets/javascripts/foreman_hosts.js [hunk 2]
```diff
@@ -31,3 +31,97 @@ export function getAttributesToPost(componentType) {
...
+    function (os) {
+      if (os[1] === '6') {
+        return [PXE_BIOS, PXE_UEFI, GRUB_UEFI];
+      } else if (os[1] === '7') {
```
Comment (lzap) [inline webpack/assets/javascripts/foreman_hosts.js]: I'd use `>= 7` here prehaps for the future.

File: webpack/assets/javascripts/foreman_hosts.test.js [hunk 3]
```diff
@@ -19,3 +19,124 @@ describe('getAttributesToPost', () => {
...
+    assertCompatibility('Debian 7.0', 'PXELinux UEFI');
+    refuteCompatibility('Debian 7.0', 'Grub UEFI');
+    assertCompatibility('Debian 7.0', 'Grub2 UEFI');
+    assertCompatibility('Debian 7.0', 'Grub2 UEFI SecureBoot');
+
+    assertUndefinedCompatibility('Debian', 'PXELinux BIOS');
+    assertUndefinedCompatibility('Debian', 'PXELinux UEFI');
+    assertUndefinedCompatibility('Debian', 'Grub UEFI');
+    assertUndefinedCompatibility('Debian', 'Grub2 UEFI');
+    assertUndefinedCompatibility('Debian', 'Grub2 UEFI SecureBoot');
```
Comment (lzap) [inline webpack/assets/javascripts/foreman_hosts.test.js:114]: Do similar tests for RHEL make sense?

**Review by lzap — CHANGES_REQUESTED** [review]
LGTM few nitpicks. I wonder if this works when PXE Loader is set via hostgroup. I will tests after rebase, if any.

File: webpack/assets/javascripts/foreman_hosts.test.js (hunk 3 above)
Comment (tstrachota) [reply webpack/assets/javascripts/foreman_hosts.test.js:114]: I added such tests for RHEL too.

**Comment by tstrachota** [comment]
@lzap updated. The warning now works also after hostgroup change.

At the moment it works only in the host form. Do you think we should add it at the hostgroup form too?

**Review by lzap — APPROVED** [review]

**Comment by lzap** [comment]
All postgresql tests failed, do you think these are relevant @tstrachota ?

**Comment by lzap** [comment]
I have two small things.

Can you please reword the text to: _This combination of loader and OS might not be able to boot._ This is shorter and no irrelevant "testing" word is used. I would drop the Warning prefix as well.

Can you add PXELoader UEFI to show warning in all cases. There is no stable build of this, we should show the warning in all cases really.

**Comment by tstrachota** [comment]
Updated, @lzap

**Comment by lzap** [comment]
Thank you, merged.

**Comment by ohadlevy** [comment]
@lzap this PR had multiple failing tests for foreman, why was it merged? develop is currently broken...

**Comment by lzap** [comment]
Sorry, it way my mistake, I was under impression this is some kind of JS random bug. I can clearly see the errors now. Thanks for fixing this @bastilian and @dLobatog and I apologize.

**Pull request merged**